src/db.rs

```diff
@@ -144,20 +147,26 @@ impl DbInner {
         &self,
         sst: &SsTableHandle,
         key: &[u8],
+        key_hash: u64,
```

Comment:
Nit: WDYT about Option<u64>, and call filter::hash_key() if it's none?

Reply:
Nit: Mind adding rustdocs to this method to clarify why we have both `key` and `key_hash` args?

Reply:
I think there is no need to change `key_hash` to Option, because we can be sure that the key in the get method is not empty. But there is no problem adding relevant comments.

Reply:
It feels slightly safer to make it an `Option` to avoid accidentally passing in the wrong value here (e.g. someone calling this method and just leaving 0 or something as a placeholder) - not sure if that's what you were worried about @criccomini.

Reply:
I'm sorry that I can't think of a scenario. If we change it to Option, we will introduce unnecessary branch and tag cost(tagged union). If we are worried that these two methods will be called incorrectly, why not turn it into closures only exist in get() method?